### PR TITLE
pass response_type param through from Flask

### PR DESCRIPTION
**Pull request by davidsbailey**
starts https://codedotorg.atlassian.net/browse/AITT-481, specifically:
* the flask endpoint in aiproxy must be updated to receive the response-type param and pass it through to the labeling codepath

other changes included:
* extract `get_example_key_concepts`, and update it to parse the examples differently based on the response type
* remove a duplicate-looking test

## Testing story

* updated unit tests in this repo
* new unit test for `assess.get_example_key_concepts`
* ran dashboard + https://github.com/code-dot-org/code-dot-org/pull/56887 against local aiproxy + this PR, and confirmed json response type is working end to end using release `2024-02-28-ai-rubrics-json-reason`

### CODE REVIEW

File: tests/routes/test_assessment_routes.py [hunk 1]
```diff
@@ -276,34 +279,6 @@ def test_should_return_400_when_the_label_function_does_not_return_the_right_str
...
-        label_mock.assert_called_with(
-            code='file data',
-            prompt='file data',
-            rubric='file data',
-            api_key=data["api-key"],
-            llm_model=data["model"],
-            remove_comments=True,
-            num_responses=2,
-            temperature=0.2
-        )
```
Comment (davidsbailey) [inline tests/routes/test_assessment_routes.py:305]: I was unable to determine if this test is covering anything substantially different from the previous test with the same name. they looked similar enough to me that I thought it was best to remove the duplicate.

Comment (davidsbailey) [reply tests/routes/test_assessment_routes.py:305]: @wilkie , do you see any reason I'm missing to keep this test around?

Comment (wilkie) [reply tests/routes/test_assessment_routes.py:305]: seems fine to remove.

**Review by wilkie — APPROVED** [review]
Looks great

**Comment by davidsbailey** [comment]
repeat manual testing revealed a bug in the code (bad merge?) and missing test coverage. I added a new unit test to address this.

**Pull request merged**